csrf_test.go

```diff
@@ -41,6 +43,49 @@ func TestProtect(t *testing.T) {
 	}
 }
 
+// TestCookieOptions is a test to make sure the middleware correctly sets cookie options
+func TestCookieOptions(t *testing.T) {
+	s := http.NewServeMux()
+	s.HandleFunc("/", testHandler)
+
+	r, err := http.NewRequest("GET", "/", nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	rr := httptest.NewRecorder()
+	expectedExpires := time.Now().Add(time.Duration(173) * time.Second)
+	p := Protect(testKey, CookieName("nameoverride"), Secure(false), HttpOnly(false), Path("/pathoverride"), Domain("domainoverride"), MaxAge(173))(s)
+	p.ServeHTTP(rr, r)
+
+	if rr.Header().Get("Set-Cookie") == "" {
+		t.Fatalf("cookie not set: got %q", rr.Header().Get("Set-Cookie"))
+	}
+
+	cookie := rr.Header().Get("Set-Cookie")
+	if strings.Contains(cookie, "HttpOnly") {
+		t.Fatalf("cookie does not respect HttpOnly option: got %v", cookie)
+	}
+	if strings.Contains(cookie, "Secure") {
+		t.Fatalf("cookie does not respect Secure option: got %v", cookie)
+	}
+	if !strings.Contains(cookie, "nameoverride=") {
+		t.Fatalf("cookie does not respect CookieName option: got %v", cookie)
+	}
+	if !strings.Contains(cookie, "Domain=domainoverride") {
+		t.Fatalf("cookie does not respect Domain option: got %v", cookie)
+	}
+	if !strings.Contains(cookie, "Max-Age=173") {
+		t.Fatalf("cookie does not respect MaxAge option: got %v", cookie)
+	}
+
+	// check for expected expire including the subsequent second to avoid edge case
+	if !strings.Contains(cookie, fmt.Sprintf("Expires=%s", expectedExpires.UTC().Format(http.TimeFormat))) &&
+		!strings.Contains(cookie, fmt.Sprintf("Expires=%s", expectedExpires.Add(time.Second).UTC().Format(http.TimeFormat))) {
+		t.Fatalf("cookie does not respect MaxAge option, unexpected expires got %v", cookie)
```

Comment:
This is failing on Go 1.4 on Travis: https://travis-ci.org/gorilla/csrf/builds/80532496

```go
-- FAIL: TestCookieOptions-2 (0.00s)
	csrf_test.go:85: cookie does not respect MaxAge option, unexpected expires got nameoverride=MTQ0MjM1NTg3M3xJbnBEWTBKdWNDdFhPVXh6TjNJdlRtbG5lVkJFYmtWM1RHOVNXU3RRVlhWc1QzUmFaWGhGWlc5RE1YYzlJZ289fIt9OI9hBudZlZ-pQwxLG-2N3AjeOyR_GhqxJGi3n_kd; Path=/pathoverride; Domain=domainoverride; Expires=Tue, 15 Sep 2015 22:27:26 UTC; Max-Age=173
```